extensions/amp-tiktok/amp-tiktok.md

```diff
@@ -0,0 +1,157 @@
+applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# amp-tiktok
+
+<!--
+  If the component is relevant for more than one format and operates differently between these
+  formats, include and filter multiple content blocks and code samples.
+-->
+
+## Usage
+
+The simplest usage of the `amp-tiktok` component has the `width`, `height` and `data-src` attributes. This will display the Tiktok embed inside of an iframe on the page.
+
+```html
+<amp-tiktok
+  width="325"
+  height="575"
+  data-src="6718335390845095173"
+></amp-tiktok>
+```
+
+`amp-tiktok` does not support autoplay and requires user interaction to play videos.
+
+### Behavior
+
+## Avoiding Layout shift
+
+The `width` and `height` attributes determine the initial laid out height of the Tiktok embedded in the page.
+
+However, we set the width of the Tiktok iframe player to be 325px wide. This results in a height of roughly 575px for the video player, however the full height of the player depends on the length and content of the Tiktok caption. (These values represent current behavior and subject to change).
+
+In order to show the entire height of the video, this component will resize its height to include the full player height. Therefore in order to avoid layout shift it is best to match the `width` and `height` attributes to those values.
```

Comment:
```suggestion
Prevent layout shift by matching the `width` and `height` attributes on the `amp-tictok` element to the size of the video player. 

By default, the width of the TikTok iframe player is set to 325px, resulting in a hight of roughly 575px for the video player. The default height of the player depends on the length and content of the TikToc caption. To show the entire video, the `amp-tiktoc` component resizes to match the player height. You can avoid this by defining the width and height to match the video. 
```

Reply:
Hopefully QQ - can you explain the relationship between `amp-tiktoc`, the TicTok video itself, the TicTok video player, and the iframe? Or is `amp-tiktoc`/TicToc video player/iframe all one in the same?

Reply:
So the `amp-tiktok` component contains an `iframe` which then itself contains the TikTok embedded player. The TikTok embedded player can be one of two aspect ratios, however in order to minimize the potential for CLS we are restricting this to the vertical aspect ratio. 

Regarding width: 
The `amp-tiktok` component can be as wide as the user wants to specify it. The `iframe` will be sized to 325px wide, and centered within the full width of the `amp-tiktok` component. 

Regarding Height:
The TikTok player resizes to its full height after its laid out. The video part of the embed is the 575px described above however the full height of the embed includes the caption space underneath. However that height is variable based on the length of the caption provided by the user. Our resize fallback resizes the component to be 775px high which we hope will encompass the full height of a TikTok with any length of caption. However, in order to prevent layout shift the developer will have to make sure that the height which they set on the `amp-tiktok` component is at minimum the final height of the TikTok embed, otherwise the resize will cause a layout shift.

Does that make sense?

Reply:
Yes, thank you! I've updated the documentation to reflect this (hopefully) better - last question, do you know how I can dynamically check the size of the TikTok player/video? Or is this done on an individual video bases?

Reply:
As far as I can tell it needs to be done on a video by video basis. The component gets the final value from the messages we receive from TikTok itself. But for sanity checks and investigation I've been just inspecting them using chrome dev tools.